src/canister_tests/src/framework.rs

```diff
@@ -175,3 +187,13 @@ where
         Err(user_error) => Err(CallError::UserError(user_error)),
     }
 }
+
+pub fn expect_user_error_with_message<T>(
+    result: Result<T, CallError>,
+    error_code: ErrorCode,
+    message: &str,
```

Comment:
nit: how about passing a `Regex` object? that'll make it clear that you can actually use regexes here